src/logic/map_objects/tribes/soldier.cc

```diff
@@ -308,9 +309,11 @@ Bob& SoldierDescr::create_object() const {
 void SoldierDescr::add_battle_animation(std::unique_ptr<LuaTable> table,
                                         SoldierAnimationsList* result) {
 	for (const std::string& anim_name : table->keys<std::string>()) {
-		if (!is_animation_known(anim_name)) {
-			throw GameDataError("Trying to add unknown battle animation: %s", anim_name.c_str());
-		}
```

Comment:
(this is not a proper review, just a quick glance, but)

Why are you removing this check? Isn't that dangerous?

Reply:
I retrieve more info about the animation through ˋget_animationˋ. In this function the same check is already done:

https://github.com/widelands/widelands/blob/master/src/logic/map_objects/map_object.cc#L393-L397